### ISSUE TITLE
Can we customize query generation?

### ISSUE THREAD

**Issue by mcun0s** [requery/requery#787]
Hi dear,

I heard that the Requery is most powerful if we use Kotlin Coroutine.
So, I'm going to use it for my new project on Kotlin.

And our database is **Apache Phoenix based on HBase**, but the Requery doesn't support officially.
I got a couple of POC for use that on our system. I think it seems like works fine, but not perfect.

The main **issue is that the 'UPSERT' command doesn't support.**
Apache Phoenix uses to that command instead of 'INSERT' and the command works for insert or update.

I think if we can do customization while make the query, then it could be resolved.
Hence, **can we customize query generator? If so, how to do that myself?**

If we can't do that, it could be painful for our project.
Thanks for your advice.

Best regards,
Unho.

**Comment by npurushe** [comment]
Yes it should be possibly by providing your database specific `Platform`, see the built in [implementations](https://github.com/requery/requery/tree/master/requery/src/main/java/io/requery/sql/platform) for examples

**Comment by mcun0s** [comment]
@npurushe thanks for your information. I'd like to make to be work Apache Phoenix for Requery.
I can share it If I have any update on that.
